velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -16,6 +16,128 @@
+
+using namespace facebook::velox;
+namespace {
+// Get values for the different supported types.
+template <typename T>
+T getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  VELOX_NYI();
+}
+
+template <>
+int8_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return static_cast<int8_t>(literal.i8());
+}
+
+template <>
+int16_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return static_cast<int16_t>(literal.i16());
+}
+
+template <>
+int32_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i32();
+}
+
+template <>
+int64_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i64();
+}
+
+template <>
+double getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.fp64();
+}
+
+template <>
+float getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.fp32();
+}
+
+template <>
+bool getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.boolean();
+}
+
+template <>
+uint32_t getLiteralValue(const ::substrait::Expression::Literal& literal) {
+  return literal.i32();
+}
+
+ArrayVectorPtr makeArrayVector(const VectorPtr& elements) {
+  BufferPtr offsets = allocateOffsets(1, elements->pool());
+  BufferPtr sizes = allocateOffsets(1, elements->pool());
+  sizes->asMutable<vector_size_t>()[0] = elements->size();
+
+  return std::make_shared<ArrayVector>(
+      elements->pool(),
+      ARRAY(elements->type()),
+      nullptr,
+      1,
+      offsets,
+      sizes,
+      elements);
+}
+
+ArrayVectorPtr makeEmptyArrayVector(memory::MemoryPool* pool) {
+  BufferPtr offsets = allocateOffsets(1, pool);
+  BufferPtr sizes = allocateOffsets(1, pool);
+  return std::make_shared<ArrayVector>(
+      pool, ARRAY(UNKNOWN()), nullptr, 1, offsets, sizes, nullptr);
+}
+
+template <TypeKind kind>
+void listLiteralToFlatVector(
+    const ::substrait::Expression::Literal& listLiteral,
+    const VectorPtr& vector,
+    vector_size_t& index) {
+  using T = typename TypeTraits<kind>::NativeType;
+  auto flatVector = vector->as<FlatVector<T>>();
+  auto typeCase = listLiteral.literal_type_case();
+  if (typeCase == ::substrait::Expression_Literal::LiteralTypeCase::kList) {
+    for (auto child : listLiteral.list().values()) {
+      listLiteralToFlatVector<kind>(child, vector, index);
+    }
+    return;
+  }
+
+  if constexpr (std::is_same_v<T, StringView>) {
+    if (typeCase == ::substrait::Expression_Literal::LiteralTypeCase::kString) {
+      flatVector->set(index++, StringView(listLiteral.string()));
+    }
+    if (typeCase ==
+        ::substrait::Expression_Literal::LiteralTypeCase::kVarChar) {
+      flatVector->set(index++, StringView(listLiteral.var_char().value()));
+    }
+  } else if constexpr (
+      std::is_same_v<T, Timestamp> || std::is_same_v<T, Date> ||
+      std::is_same_v<T, IntervalDayTime>) {
```

Comment:
Why do we do nothing if type is Timestamp or IntervalDayTime? Let's add test cases to make sure these types are working properly.